addons/block_code/serialization/block_serialization.gd

```diff
@@ -11,3 +13,20 @@ func _init(p_name: StringName = &"", p_children: Array[BlockSerialization] = [],
 	name = p_name
 	children = p_children
 	arguments = p_arguments
+
+
+# Block name backwards compatibility handling.
+const _renamed_blocks: Dictionary = {
+	&"simplespawner_set_spawn_frequency": &"simplespawner_set_spawn_period",
+}
+
+
+func _set_name(value):
+	var new_name = _renamed_blocks.get(value)
+	if new_name:
+		print("Migrating block %s to new name %s" % [value, new_name])
+		name = new_name
+		if Engine.is_editor_hint():
+			EditorInterface.mark_scene_as_unsaved()
+	else:
+		name = value
```

Comment:
I'm happy with just this as a place to start. Two thoughts for the distant future:

- There's a danger that the old name is now unusable in this case, so we might want to couple it with a version number or something.
- This requires that we maintain a list of renamed blocks in the core plugin code. That's an issue for blocks added outside of the core plugin. Those aren't a thing at the moment, but a different (albeit fussier) approach could be a `renamed_from` field in block_definition.gd.

Reply:
Indeed, this only handles pure renames. If you changed the arguments or something like that, this wouldn't work. We do have a version number [in the higher level BlockScriptSerialization](https://github.com/endlessm/godot-block-coding/blob/2f66ee7f9489deace3becce1ba0478281bdaad81/addons/block_code/serialization/block_script_serialization.gd#L29), but it's not available here right now.

@wjt poked at adding aliases for blocks in #284. I didn't look closely at his approach there, but it was my initial suggestion :laughing:

Reply:
Unfortunately I actually did change the argument name in the "set ~~frequency~~ period" block...